app/forms/household_composition_form.py

```diff
@@ -0,0 +1,112 @@
+from flask_wtf import FlaskForm
+from wtforms import FieldList, Form, FormField, StringField
+from wtforms import validators
+
+from app.data_model.answer_store import Answer
+from app.helpers.schema_helper import SchemaHelper
+
+from werkzeug.datastructures import MultiDict
+
+
+def get_name_form(block_json, error_messages):
+    class NameForm(Form):
+        pass
+
+    for answer in SchemaHelper.get_answers_for_block(block_json):
+        validate_with = [validators.optional()]
+
+        if answer['mandatory'] is True:
+            error_message = error_messages['MANDATORY']
+            if 'validation' in answer and 'messages' in answer['validation'] \
+                    and 'MANDATORY' in answer['validation']['messages']:
+                error_message = answer['validation']['messages']['MANDATORY']
+
+            validate_with = [validators.InputRequired(
+                message=error_message,
+            )]
+
+        # Have to be set this way given hyphenated names
+        # are considered invalid in python
+        field = StringField(validators=validate_with)
+
+        setattr(NameForm, answer['id'], field)
+
+    return NameForm
+
+
+def serialise_composition_answers(location, data):
```

Comment:
The error structure isn't customisable and depends on the form structure, so you couldn't manipulate it's structure through using form state. The forms state is populated using the data parameter supplied to the generate method.